lightning_pose/data/utils.py

```diff
@@ -395,18 +395,11 @@ def generate_heatmaps(
     confidence *= -1
     confidence /= 2 * sigma**2
     confidence = torch.exp(confidence)
-
-    # replace nans with uniform heatmap
-    uniform_heatmap = torch.ones(
-        (out_height, out_width), device=keypoints.device) / (out_height * out_width)
-    confidence[nan_idxs] = uniform_heatmap
-
-    if normalize:
-        # normalize all heatmaps to one
-        confidence = confidence / torch.sum(confidence, dim=(2, 3), keepdim=True)
-    else:
-        confidence /= sigma * torch.sqrt(2 * torch.tensor(np.pi))
-
+    # normalize all heatmaps to one
+    confidence = confidence / torch.sum(confidence, dim=(2, 3), keepdim=True)
+    # replace nans with zeros heatmaps
```

Comment:
add a comment that "we won't try to output an all-zeros heatmap, but rather all-zeros heatmaps will be ignored in the supervised heatmap loss"